### PR TITLE
update to use ibm-clang or clang

### PR DESCRIPTION
**Pull request by netskink**

### CODE REVIEW

**Comment by IgorTodorovskiIBM** [comment]
I would prefer if it checks for the presence of ibm-clang, and if it exists in your path, then it will use, else check for clang. This logic can be applied to zopen_append_to_env()

**Comment by MikeFultonDev** [comment]
> I would prefer if it checks for the presence of ibm-clang, and if it exists in your path, then it will use, else check for clang. This logic can be applied to zopen_append_to_env()

This makes sense to me. If there is both a supported and unsupported version of clang available, I would think people will want the supported one.

**Comment by netskink** [comment]
> I would prefer if it checks for the presence of ibm-clang, and if it exists in your path, then it will use, else check for clang. This logic can be applied to zopen_append_to_env()

Sounds good.  So, use ibm-clang if it exists and fallback to xlclang if it doesn't.   Put mod in zopen_append_to_env().  I'll get right on it.  Thanks for the pointer!

**Comment by netskink** [comment]
fwiw, the above second revision is an effort to meet Igor's objectives.

File: buildenv [hunk 1]
```diff
@@ -26,12 +27,55 @@ zopen_append_to_env()
...
+        if [ "\${CLANG_BIN}" = "clang" ] ; then
+            # clang exists
+            echo "Using clang: \${CLANG_PATH}"
+        else
+            echo "clang does not exist, trying next compiler choice..."
+            CLANG_BIN=""
+        fi
+    fi
+
+    # lastly try xlclang
```
Comment (IgorTodorovskiIBM) [inline buildenv:58]: I suggest removing xlclang fallback because the options sets are vastly different and there's also a xlclang_port for it.

Comment (netskink) [reply buildenv:58]: sounds good.  Will do.  Thanks!  I'll address in morning.

**Comment by netskink** [comment]
The above commit, removes xlclang.  This is good to know there is another comp_xxx wrapper.  I also, adjusted so that in case a compiler is not found, it will pass to the install message.

**Review by IgorTodorovskiIBM — APPROVED** [review]
Looks good, thank you!

**Comment by netskink** [comment]
You are welcome.  By the way, I'm in the zotsample repo but not this one. I can not merge it.

**Pull request merged**